ratelimits/utilities_test.go

```diff
@@ -0,0 +1,27 @@
+package ratelimits
+
+import (
+	"testing"
+
+	"github.com/letsencrypt/boulder/test"
+)
+
+func TestDomainsForRateLimiting(t *testing.T) {
+	domains := DomainsForRateLimiting([]string{})
+	test.AssertEquals(t, len(domains), 0)
+
+	domains = DomainsForRateLimiting([]string{"www.example.com", "example.com"})
+	test.AssertDeepEquals(t, domains, []string{"example.com"})
+
+	domains = DomainsForRateLimiting([]string{"www.example.com", "example.com", "www.example.co.uk"})
+	test.AssertDeepEquals(t, domains, []string{"example.co.uk", "example.com"})
+
+	domains = DomainsForRateLimiting([]string{"www.example.com", "example.com", "www.example.co.uk", "co.uk"})
+	test.AssertDeepEquals(t, domains, []string{"co.uk", "example.co.uk", "example.com"})
```

Comment:
I'd break the inclusion of "co.uk" out into a separate test case, since it's a surprising edge case.

Also, all of these tests could have comments indicating which behavior they're intending to exhibit.

Reply:
Ah, these actually came over from the RA and I should have been omitted from this changeset since the Bucket constructors which use it are being shipped in the PR adding key-value NewOrder limits to the WFE.

Reply:
I can either leave them as-is (which will shorten up the last PR) or I can remove them from this PR.

Reply:
Ah you're right, I hadn't realized that `DomainsForRateLimiting` isn't actually called yet. Let's leave it out until it's called, so we're not landing dead code.